### ISSUE TITLE
 Performance issues in tpu/models/official/retinanet/dataloader.py

### ISSUE THREAD

**Issue by DLPerf** [richardaecn/class-balanced-loss#20]
Hello,I found a performance issue in the definition of `__call__` ,
tpu/models/official/retinanet/dataloader.py,
[dataset = dataset.map(_process_example)](https://github.com/richardaecn/class-balanced-loss/blob/1d7857208a2abc03d84e35a9d5383af8225d4b4d/tpu/models/official/retinanet/dataloader.py#L379) was called without **num_parallel_calls**.
I think it will increase the efficiency of your program if you add this.

The same issues also exist in [dataset = dataset.map(parser)](https://github.com/richardaecn/class-balanced-loss/blob/1d7857208a2abc03d84e35a9d5383af8225d4b4d/tpu/models/experimental/dcgan/mnist_input.py#L68) ,
[dataset = dataset.repeat().map(parser)](https://github.com/richardaecn/class-balanced-loss/blob/1d7857208a2abc03d84e35a9d5383af8225d4b4d/tpu/models/experimental/inception/inception_v3_old.py#L256)

Here is [the documemtation of tensorflow](https://tensorflow.google.cn/api_docs/python/tf/data/Dataset?hl=en#map) to support this thing.

Looking forward to your reply. Btw, I am very glad to create a PR to fix it if you are too busy.
